game-core/src/main/java/games/strategy/triplea/delegate/AAInMoveUtil.java

```diff
@@ -236,13 +236,11 @@ private PlayerID movingPlayer(final Collection<Unit> units) {
       return player;
     }
 
-    for (final Unit u : units) {
-      if (u != null && u.getOwner() != null) {
-        return u.getOwner();
-      }
-    }
-
-    return PlayerID.NULL_PLAYERID;
+    return units.stream()
+        .filter(u -> u != null && u.getOwner() != null)
+        .findFirst()
+        .map(Unit::getOwner)
```

Comment:
```java
.filter(Objects::nonNull)
.map(Unit::getOwner)
.filter(Objects::nonNull)
.findAny()
```

Reply:
that does not compile @RoiEXLab 

One question, why does the unit list have any null elements at all? I don't think we should spend more time at this point on better streaming syntax, there is a more major consistency/correctness problem, most of the filtering for nulls should not be necessary.

I"m getting concerned the scope can get out of hand if I start to fixing the real problems, the better syntax seems like band-aid :(